### DISCUSSION TITLE
Suggestions stop working in markdown file with this code in the code fence

### DISCUSSION THREAD

**Discussion by TheColorRed** [community/community#49002] [Copilot Conversations]
### Select Topic Area

Product Feedback

VSCode: 1.76.0
GitHub Copilot: v1.76.9071

### Body

I am writing documentation in a `.md` file for one of my companies products which is written in Angular. I seem to stop getting suggestions when with this in the doc:

```md

# Example

This is an example:

\```
@Component({
  standalone: true,
  imports: [ConfirmationModule, ButtonModule, BrowserAnimationsModule],
  selector: 'confirmation-example',
  template: `
    <button (click)="showConfirmation()">
      Open Confirmation
    </button>
  `,
})
class ConfirmationExample {
  constructor(private readonly modal: ModalService) { }
  showConfirmation() {
    this.modal.create(ConfirmationComponent, {
      status: 'error',
      data: {
        title: 'Confirmation',
        message: 'Are you sure you want to do this?',
      },
      size: 'small',
      align: 'center',
    });
  }
}
\```
```

**Note:** Not sure how to display a code fence within a code fence, so I just escaped it (those slashes aren't actually there in the markdown file).

When I remove this section, suggestions start working again:

```
      data: {
        title: 'Confirmation',
        message: 'Are you sure you want to do this?',
      }
```

One of two message show up in the `GitHub Copilot` output:

```
[INFO] [default] [2023-03-02T17:27:38.840Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-03-02T17:27:38.946Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 106 ms
[INFO] [streamChoices] [2023-03-02T17:27:38.948Z] request done: headerRequestId: [] model deployment ID: []
```

```
[INFO] [streamChoices] [2023-03-02T17:26:58.686Z] request done: headerRequestId: [721f1deb-a936-4755-9c37-68dd47430047] model deployment ID: [x625ba4f78aa8]
[INFO] [ghostText] [2023-03-02T17:27:36.978Z] Cancelled by contextual filter
```

### Edit

After playing around for a bit I see if i move the `data` part as the last item in the object, that also helps get the suggestions working again.

**Comment by TheColorRed** [comment]
I have added another file, it also seems to not like it. When I write this:

```md
# Conclusion

This tutorial has shown you how to create Angular components in a library that supports tree shaking. You should now know how to make your own library using the same techniques.
```

I no longer get suggestions. If I remove the paragraph, suggestions start working again.

**Comment by chocotan** [comment]
I have the same issue until I deleted all lines in my java class file.
I removed and reinstall the plugin but it doesn't work.
![image](https://user-images.githubusercontent.com/1758565/229134014-c987a199-9db1-456d-be68-425ea1bf4cd8.png)
![image](https://user-images.githubusercontent.com/1758565/229134285-4939da8a-d8e3-4e06-95bd-67e2f1834ae0.png)

**Reply by kming852** [reply]
seem to have the same exact issue, have you figured out why?

**Comment by bytesbay** [comment]
Have the same for src/* path in my project ...

**Comment by Nick6464** [comment]
I had the same issue in a front end, but  not the backend, so it wasn't a config error for me. 
I found when I deleted huge amounts of code it would work sometimes. I narrowed it down to one line.
"Something looks (f word)y here..." For an error message. on the removal of the f word, I started getting suggestions. It seems like Copilot has content filtering that is strict since its a completion which would require it output potentially offensive content.

Hope this helps someone.

**Reply by reihig-ut** [reply]
I have been struggling to find the cause for a long long time, but finally solved the problem thanks to you. Thank you so much.
In my case, it seems that what I was using as an abbreviation for "assumption" was not good :cry:

**Reply by lloydtripp** [reply]
I have this same issue! It took me so long to figure out what's going. Had you not suggested what's going I would have never figured it out. 

Turns out, Github thinks I'm using offensive language despite it being for biological research. Not sure what to do to get around this.
![Screenshot 2024-01-28 174017](https://github.com/community/community/assets/6070112/92f2d7a4-846e-4410-8018-929d58e7fee4)
![Screenshot 2024-01-28 174100](https://github.com/community/community/assets/6070112/fc3cfe71-891e-48f8-bba2-cb28a79391ff)

**Reply by merely-present** [reply]
I think I just discovered that it also doesn't like "spics_io_num" from the esp idf, which is a pretty annoying thing to arbitrarily completely halt over. Are unreliable tools really tools, is the question.

**Comment by lailo** [comment]
In my case, the issue was a value in the `settings.json` file of my VSCode settings. The value `live3` in  `"inlineChat.mode": "live3"` caused the issue, and when removed, it worked again.

**Comment by darren277** [comment]
I just spent a bit of time this morning troubleshooting what I can only imagine is the same problem.

I was actually creating a new discussion for it and then at the end of it, clicked the "I searched for an existing discussion about this" link at the end and found this.

I was so perplexed as to how I was getting no completions across different editors and different versions of the same editors and after updating and reinstalling this, that, and the other.

It did not occur to me that it could simply be the specific file that's the issue.

It would be **tremendously** helpful if there could be some kind of feedback indicating that CoPilot detected a cuss word, perceived or otherwise, and therefore won't work for this particular file / code snippet.

___

# The model deployment ID (from the CoPilot logs) is empty

All of a sudden CoPilot has started behaving strangely for me across two different editors (VSCode and PyCharm).

I've of course tried the usual things like disabling it and re-enabling it, uninstalling it and re-installing it. Restarting the editors. Updating the editors. The works.

And then I noticed it's still working on one instance of VSCode I have running.

I compared the CoPilot logs and did find an interesting discrepancy.

## WORKING CASE:
```
2024-02-28 07:20:14.047 [info] [streamChoices] request done: headerRequestId: [25a18817-3647-4e6d-833d-41e67bc18ece] model deployment ID: [y42acfba7506a]
2024-02-28 07:20:14.942 [info] [fetchCompletions] request.response: [[https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions]](vscode-file://vscode-app/c:/Users/Darren/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) took 118 ms
```

## BROKEN CASE:
```
2024-02-28 07:18:00.285 [info] [streamChoices] request done: headerRequestId: [6489de70-b293-4b55-8671-907a1d3f7035] model deployment ID: []
2024-02-28 07:18:01.575 [info] [fetchCompletions] request.response: [[https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions]](vscode-file://vscode-app/c:/Users/Darren/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) took 132 ms
```

## The difference
That model deployment ID is empty in the broken case.

I'm not really sure what to make of it but figured it would be worth bringing up.


## Addendum after searching for similar discussions
It turns out that I'm not the first person to discover this particular problem.

**Reply by lloydtripp** [reply]
https://www.theregister.com/2021/09/02/github_copilot_banned_words_cracked/

**Reply by petertheprocess** [reply]
So, any solution about that?

**Reply by darren277** [reply]
> So, any solution about that?

@petertheprocess, if the issue is a banned word, the only solution, from what I can tell, is to either remove it, rename it, cens*r it, or temporarily relocate it until you no longer need to use Co-Pilot when inside of that particular file.

Of course, the hard part is finding out that's the root cause in the first place, since there are no helpful errors being shown in whichever code editor you're using Co-Pilot with, or even being logged to the console if you're trying to debug it yourself.

If you already know you've got a no-no word (according to Co-Pilot's sensibilities), the hard part is over. All you have to do - and it seems to be your only option, really - is to make it no longer be a no-no word.
